### ISSUE TITLE
Timeout connecting to server (remote.php/webdav)

### ISSUE THREAD

**Issue by janpekar** [nextcloud/desktop#1209]
I'm using nextcloud linux client and suddenly it stops connecting. It stops with error 
"No connection to Nextcloud at URL" and "Connection timed out"
 
I was debugging problem and notice, that ```OCC::PropfindJob```
which is connecting to /remote.php/webdav/ reports network timeout error after aprox. 25-30 seconds, but apache serves that URL after 120 or more seconds in my case. For other users it takes from 2-15 seconds so they have no problem with that.
 
I'm sharing a lot of files and lot of changes happens on the server and client.

I tried to change timeout value in .config/Nextcloud/nextcloud.cfg but it didn't help (it is ignoring settings). Also default should be 300 seconds which is not what I get when I was measuring the time during parsing nextcloud client logs.

I tried client 2.5.2 and server is 15.0.7

I'm pasting log from client. Problem occurs after line ```OCC::AbstractNetworkJob::start         OCC::PropfindJob created for ...``` and after approx. 25 seconds next line reports problem (much sooner, than remote.php finishes.

```
[OCC::ConnectionValidator::slotStatusFound      ** Application: ownCloud found:  QUrl("https://xxxxxxxxxxxxxxxx")  with version  "15.0.7" ( "15.0.7.0" )
[OCC::ConnectionValidator::setAndCheckServerVersion     QUrl("https://xxxxxxxxxxxxxxxx") has server version "15.0.7.0"
[OCC::AccessManager::createRequest      6 "PROPFIND" "https://xxxxxxxxxxxxxxxx/remote.php/webdav/" has X-Request-ID "7ecd28a2-7f58-4db7-9ad6-74e9896c6eee"
[OCC::AbstractNetworkJob::start         OCC::PropfindJob created for "https://xxxxxxxxxxxxxxxx" + "/" "OCC::ConnectionValidator"
[OCC::AbstractNetworkJob::slotTimeout   Network job timeout QUrl("https://xxxxxxxxxxxxxxxx/remote.php/webdav/")
[OCC::WebFlowCredentials::slotFinished  request finished
[OCC::AbstractNetworkJob::slotFinished  QNetworkReply::NetworkError(OperationCanceledError) "Connection timed out" QVariant(Invalid)
[OCC::WebFlowCredentials::stillValid    QNetworkReply::NetworkError(OperationCanceledError)
[OCC::WebFlowCredentials::stillValid    "Operation canceled"
[OCC::PropfindJob::finished     PROPFIND of QUrl("https://xxxxxxxxxxxxxxxx/remote.php/webdav/") FINISHED WITH STATUS "OperationCanceledError Connection timed out"
[OCC::PropfindJob::finished     *not* successful, http result code is 0 ""
[OCC::WebFlowCredentials::stillValid    QNetworkReply::NetworkError(OperationCanceledError)
[OCC::WebFlowCredentials::stillValid    "Operation canceled"
[unknown        QIODevice::read (QNetworkReplyHttpImpl): device not open
[OCC::FolderMan::slotAccountStateChanged        Account "jan.pekar@xxxxxxxxxxxxxxxx" disconnected or paused, terminating or descheduling sync folders
```
Thank you for help.

**Comment by janpekar** [comment]
After disconnecting and connecting to the Internet I have same problem  - server request takes too long and nextcloud client reports timeout.

I fixed it by optimizing sql database on the server.

**Comment by camilasan** [comment]
Shall I close the issue @janpekar?

**Comment by janpekar** [comment]
I think that client still needs to be fixed to obey network timeout from config file.

On the other way - now I don't need that so if I'm the only one with slow SQL server, you can close the issue.

**Comment by camilasan** [comment]
Could you check how it works for you with 2.5.3 rc1?
https://download.nextcloud.com/desktop/prereleases/Linux/Nextcloud-2.5.3.20190412-rc1-x86_64.AppImage
Thanks!

**Comment by loxK** [comment]
I have the same issue, I can upload 1.5GB easily in less than a minute using the webui.

I tried 2.5.3 RC1 , it is worst, I get an error for myfile.mp4 of 14mb: "File name with //Vidal"

```
#=#=#=#=# Propagation starts 2019-05-08T05:24:41Z (last step: 680 msec, total: 680 msec)
||Vidéos/myfile.mp4|INST_SYNC|Up|1557210810|1de13e56a8e6d3ce446f5878091c8c94|15024931|04394721ocxqcnu4i3nc|2|Error transferring https://[redacted].com/nextcloud/remote.php/dav/uploads/laurent/3555657213/.file - server replied:  (File with name //Vidal could not be located)|404|476022|1557210379||||
#=#=#=# Syncrun finished 2019-05-08T05:24:42Z (last step: 541 msec, total: 1221 msec)
#=#=#=# Syncrun started 2019-05-08T05:24:57Z
#=#=#=#=# Propagation starts 2019-05-08T05:24:57Z (last step: 267 msec, total: 267 msec)
#=#=#=# Syncrun finished 2019-05-08T05:24:57Z (last step: 109 msec, total: 377 msec)
#=#=#=# Syncrun started 2019-05-08T05:25:47Z
#=#=#=#=# Propagation starts 2019-05-08T05:25:47Z (last step: 329 msec, total: 329 msec)
||Vidéos/Challenge Newsletter.mp4|INST_SYNC|Up|1557210810|1de13e56a8e6d3ce446f5878091c8c94|15024931|04394721ocxqcnu4i3nc|2|Error transferring https://[redacted].com/nextcloud/remote.php/dav/uploads/laurent/2518168235/.file - server replied:  (File with name //Vidal could not be located)|404|476022|1557210379||||
#=#=#=# Syncrun finished 2019-05-08T05:25:48Z (last step: 536 msec, total: 865 msec)
```

**Comment by NicolasGoeddel** [comment]
I am using version 2.6.0git and this issue still exists. After starting the system from suspend most of the time the nextcloud-client tells me about the timeout and does nothing more. Normally I would it expect to retry after a while but it does nothing. The webinterface of Nextcloud is snappy as always and webdav is also connected.

After closing the client and starting it again it takes a while and the timeout occurs again. I don't know what I could do about it and from where I could give you more details about the issue, but it definitely annoys me. It would be nice to solve that issue soon. I don't understand why it's sometimes works and sometimes not.

**Comment by asonix** [comment]
I'm having this same issue with `Nextcloud-3.1.2-x86_64.Appimage`. Previously, I was using the 3.1.2 flatpak image, and I noticed that my folders hadn't seemed to sync for a few days. I tried removing all the folders I had synced and re-adding them, and then removing the account and re-adding it, and neither thing helped.

My issue manifests as the client moving to `Preparing to sync...`, and sitting there for a while as it issues PROPFIND requests for seemingly every directory in my server, even though I'm only syncing a single subfolder. After a short time, it shows a `No connection to Nextcloud at [my server]. Connection timed out` error, then moves to `Waiting ...` and never seems to try again, even if I manually tell it to restart the sync. If I completely close the desktop client and restart it, the sync will be attempted again with the same result.

My server is currently running 20.0.7 on a rockpro64 (aarch64 sbc), with a postgres 13 db on a separate rock64 (slightly less powerful aarch64 sbc).

There seem to be a few issues at work here
1. The client issues PROPFIND requests for all folders on the server, not just folders that will be synced.
2. The connection timeout happens in just under 1 minute from the start of the sync, seemingly ignoring the timeout setting in the nextcloud config.
3. A single connection timeout causes the sync to abort, even when other in-flight requests complete successfully after the timeout error is received. 
4. After a connection timeout, sync seems to never be attempted again.
5. Manually restarting a sync after a connection timeout seemingly does nothing.

**Comment by janpekar** [comment]
As I mentioned before. My "solution" was database optimization so that timeout didn't occur. The issue may still exists in the client, but it will not affect you.

Do you have all recommended nextcloud sql upgrades? ```occ db: add-missing-indices```
How about postgresql optimalization, some pg tune config (https://pgtune.leopard.in.ua)
Speed-up iops by placing db to SSD etc..

**Comment by asonix** [comment]
I have run the nextcloud sql upgrades, and I've applied the pgtune settings as well. I'm running the database off an SSD. I also set up redis distributed caching and APCu local caching.

Even after all of this, I'm still encountering this problem.

**Comment by asonix** [comment]
For additional context, I'm hosting about 300GB on my server, and this problem only started occurring recently

**Comment by er-vin** [comment]
>     1. The client issues PROPFIND requests for all folders on the server, not just folders that will be synced.

Do you have e2ee enabled on that server? We found a performance issue in that code path which could lead to similar situations on very large installs. If that's the case then your issue should be fixed with 3.2.0 and is different from what @janpekar reported back in 2019.

**Comment by asonix** [comment]
I do, although I don't currently sync any e2ee files. Good to hear that this might be fixed!
